File: csharp/parallel-letter-frequency/ParallelLetterFrequency.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using System.Collections.Concurrent;
using System.Linq;

public static class ParallelLetterFrequency
{
    public static IDictionary<char, int> Calculate(IEnumerable<string> texts)
    {
        var cd = new ConcurrentDictionary<char, int>();
        Parallel.ForEach(texts, s =>
            Parallel.ForEach(s.ToLower().Where(c => Char.IsLetter(c)), c =>
                cd.AddOrUpdate(c, 1, (key, oldValue) => oldValue + 1)
            )
        );
        return cd;
    }
}

File: fsharp/parallel-letter-frequency/ParallelLetterFrequency.fs
module ParallelLetterFrequency

open System

let frequency (texts: string list) = 
    texts 
    |> List.toArray
    |> Array.Parallel.collect (fun text -> 
        text.ToLower().ToCharArray()
        |> Array.filter (Char.IsLetter))
    |> Array.groupBy id
    |> Array.Parallel.map (fun (key, arr) -> (key, arr |> Array.length) )
    |> Map.ofArray 
        